### ISSUE TITLE
dotnet segfaults on Debian 9 (Stretch)

### ISSUE THREAD

**Issue by Yutsa** [dotnet/cli#6542]
## Steps to reproduce
Follow [these install instructions](https://www.microsoft.com/net/core#linuxdebian) then I followed [these](https://github.com/dotnet/cli/issues/2018#issuecomment-239301891) instructions to be able to execute `dotnet`.

Now it segfaults when using `dotnet new console -o hwapp` and it also failed with `dotnet restore` and `dotnet run`

## Expected  behavior

It should create a new project.

## Actual behavior

It segfaults.

## Environment data
`dotnet --info` output:

```
$ dotnet --info
.NET Command Line Tools (1.0.3)

Product Information:
 Version:            1.0.3
 Commit SHA-1 hash:  37224c9917

Runtime Environment:
 OS Name:     debian
 OS Version:  9
 OS Platform: Linux
 RID:         debian.8-x64
 Base Path:   /opt/dotnet/sdk/1.0.3
```

```
$ uname -a
Linux edouard 4.9.0-2-amd64 #1 SMP Debian 4.9.18-1 (2017-03-30) x86_64 GNU/Linux
```

`gdb` output : 

```
(gdb) run dotnet new console -o hwapp
Starting program: /usr/local/bin/dotnet dotnet new console -o hwapp
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7ffff4789700 (LWP 11590)]
[New Thread 0x7ffff3f88700 (LWP 11591)]
[New Thread 0x7ffff3787700 (LWP 11592)]
[New Thread 0x7ffff2f86700 (LWP 11593)]
[New Thread 0x7ffff2785700 (LWP 11594)]
[New Thread 0x7ffff1c0b700 (LWP 11595)]
[New Thread 0x7fffe2135700 (LWP 11596)]
[New Thread 0x7fffe1934700 (LWP 11597)]
[New Thread 0x7fffe1133700 (LWP 11598)]
[New Thread 0x7fffe0932700 (LWP 11599)]
[New Thread 0x7fffe0131700 (LWP 11600)]
[New Thread 0x7fffdf930700 (LWP 11601)]
No executable found matching command "dotnet-dotnet"
[Thread 0x7fffdf930700 (LWP 11601) exited]
[Thread 0x7fffe0932700 (LWP 11599) exited]
[Thread 0x7fffe1133700 (LWP 11598) exited]
[Thread 0x7fffe1934700 (LWP 11597) exited]
[Thread 0x7fffe2135700 (LWP 11596) exited]
[Thread 0x7ffff1c0b700 (LWP 11595) exited]
[Thread 0x7ffff2785700 (LWP 11594) exited]
[Thread 0x7ffff2f86700 (LWP 11593) exited]
[Thread 0x7ffff3787700 (LWP 11592) exited]
[Thread 0x7ffff3f88700 (LWP 11591) exited]
[Thread 0x7ffff4789700 (LWP 11590) exited]
[Thread 0x7ffff7fbf740 (LWP 11586) exited]
[Inferior 1 (process 11586) exited with code 01]
(gdb) 
```

This is what I gathered for now. @karelz and @janvorli if you need more info I'll try to provide them.

**Comment by karelz** [comment]
@janvorli any suggestion how to make progress? I guess we need some callstack or something ...

**Comment by Yutsa** [comment]
Actually reverting to Jessie's version of `libcurl3` fixed the issue ! It was suggested [here](https://github.com/dotnet/corefx/issues/8951#issuecomment-299927277).

**Comment by v6ak** [comment]
I have the same issue with Debian Stretch (stable version at the moment). Commands `dotnet restore `, `dotnet migrate` and `dotnet new console -o hwapp` segfault. 

Downgrading libcurl3 might be a workaround, but it might have some side effects.

I've looked for missing libraries using `find /opt/dotnet -name '*.so' -type f -print | xargs ldd | grep 'not found'`. I was able to fix one of them by installing liblttng-ust0/stable, but I cannot get liblldb-3.6.so, because there are only newer versions (3.8 and 3.9) of the library.

    % uname -a
    Linux ysoft 4.9.28-16.pvops.qubes.x86_64 #1 SMP Tue May 16 00:38:09 UTC 2017 x86_64 GNU/Linux
